main/main.cpp

```diff
@@ -35,62 +36,127 @@ static void setup_nvs_flash() {
   ESP_ERROR_CHECK(ret);
 }
 
+static std::string serialize_localization_targets(
+    const std::vector<LocalizationTarget> &localization_targets,
+    int tracker_id) {
+  auto *json_object = cJSON_CreateObject();
+  cJSON_AddNumberToObject(json_object, "trackerId", tracker_id);
+  auto *json_measurements = cJSON_AddArrayToObject(json_object, "measurements");
+
+  for (const auto &target : localization_targets) {
+    const auto &address = target.m_mac_address;
+    char mac_address[128];
+    snprintf(mac_address, sizeof(mac_address), "%02x:%02x:%02x:%02x:%02x:%02x",
+             address.bytes[0], address.bytes[1], address.bytes[2],
+             address.bytes[3], address.bytes[4], address.bytes[5]);
+
+    auto *json_measurement_item = cJSON_CreateObject();
+    cJSON_AddStringToObject(json_measurement_item, "macAddress", mac_address);
+    cJSON_AddNumberToObject(json_measurement_item, "type",
+                            (int)target.m_sniffer_type);
+    cJSON_AddNumberToObject(json_measurement_item, "rssi", target.m_rssi);
+
+    cJSON_AddItemToArray(json_measurements, json_measurement_item);
+  }
+  auto *json_string = cJSON_Print(json_object);
+  std::string json_data{json_string};
+  free(json_string);
+
+  cJSON_Delete(json_object);
+
+  return json_data;
+}
+
 extern "C" void app_main(void) {
   setup_nvs_flash();
 
   esp_event_loop_create_default();
 
+  const auto wifi_mac_address = WifiConnection::get_mac_address();
+  const auto bluetooth_mac_address = BluetoothClassicScanner::get_mac_address();
+
   WifiConnection::setup_wifi();
 
   while (true) {
+    const auto freeHeap = xPortGetFreeHeapSize();
+    ESP_LOGI("main", "Free heap: %zuKB", freeHeap / 1000);
+    bool registered = false;
+    int tracker_id = -1;
+
     {
       WifiConnection connection;
 
-      {
-        HttpRequest createRoomRequest("http://10.0.0.1:8081/room/add",
-                                      HTTP_METHOD_POST);
-        const auto response = createRoomRequest.send_with_post_data(
-            R"(
-          { "RoomId": 0,
-            "Name": "Created from esp",
-            "Description": "Fuck you!"
-          })");
-        printf("%s\n", response.data());
-      }
-
-      HttpRequest request{"http://10.0.0.1:8081/room/all", HTTP_METHOD_GET};
-      const auto str = request.send();
-      printf("%s\n", str.data());
+      do {
+        char url_string[256];
+
+        snprintf(
+            url_string, sizeof(url_string),
+            "http://10.0.0.1:8081/tracker/registration/info"
```

Comment:
Hardcoded IP address, but lets be real, it will always be 10.0.01 in our case.
However, incase we need more servers and if the trackers are on different networks, this needs to be changed